### PR TITLE
Ci managed apps

### PR DESCRIPTION
**Pull request by postwait**

### CODE REVIEW

File: src/utils/mtev_watchdog.c [hunk 1]
```diff
@@ -192,6 +193,115 @@ typedef enum {
...
+  int i;
+  int idx = nmanaged++;
+  mtevAssert(manage_allowed);
+  managed = realloc(managed, sizeof(*managed) * nmanaged);
+  memset(&managed[idx], 0, sizeof(*managed));
+  managed[idx].file = strdup(file);
+  for(i=0;argv[i] != NULL;i++);
+  managed[idx].argv = calloc(i+1, sizeof(char *));
+  for(i=0;argv[i] != NULL;i++) managed[idx].argv[i] = strdup(argv[i]);
+  managed[idx].argv[i] = NULL;
```
Comment (cody271) [inline src/utils/mtev_watchdog.c]:
```suggestion
  managed[idx].argv[i+1] = NULL;
```

File: src/utils/mtev_watchdog.c [hunk 2]
```diff
@@ -192,6 +193,115 @@ typedef enum {
...
+  for(i=0;envp[i] != NULL;i++);
+  managed[idx].envp = calloc(i+1, sizeof(char *));
+  for(i=0;envp[i] != NULL;i++) managed[idx].envp[i] = strdup(envp[i]);
+  managed[idx].envp[i] = NULL;
```
Comment (cody271) [inline src/utils/mtev_watchdog.c]:
```suggestion
  managed[idx].envp[i+1] = NULL;
```

File: src/utils/mtev_watchdog.c [hunk 3]
```diff
@@ -192,6 +193,115 @@ typedef enum {
...
+      mtev_log_stream_pipe_t *out2 = mtev_log_stream_pipe_new(m->err);
+      m->pid = fork();
+      if(m->pid == -1) {
+        mtevL(mtev_error, "[monitor] fork error: %s\n", strerror(errno));
+        mtev_log_stream_pipe_close(out1);
+        mtev_log_stream_pipe_close(out2);
+        m->pid = 0;
+      }
+      else if(m->pid == 0) {
+        setpgid(0, 0);
```
Comment (cody271) [inline src/utils/mtev_watchdog.c]:
```suggestion
        if(setpgid(0, 0) != 0) {
          mtevFatal(mtev_error, "setpgid() failed %s\n", strerror(errno));
        }
```

File: src/utils/mtev_log.c [hunk 4]
```diff
@@ -2428,6 +2429,192 @@ mtev_log_stream_to_json(mtev_log_stream_t ls) {
...
+    /* Setup polling */
+    pthread_mutex_lock(&logpipes_lock);
+    if(!head) {
+      mtevL(mtev_debug, "logpipes consumer exiting\n");
+      pthread_mutex_unlock(&logpipes_lock);
+      break;
+    }
+    int i = 0;
+    for(mtev_log_stream_pipe_t *lp = head; lp; lp = lp->next) i++;
+    if(i > nfds) {
```
[the remark this answers is not in the document]
Comment (postwait) [reply src/utils/mtev_log.c]: No nfds can shrink down as well.  Each tim through the loop it resizes to the number of active logpipes.

File: src/utils/mtev_watchdog.c (hunk 2 above)
Comment (postwait) [reply src/utils/mtev_watchdog.c]: that's not right. we leave the loop with [i] unset.  It should be `i` not `i+1`

**Pull request merged**